homeassistant/components/fmi/__init__.py

```diff
@@ -0,0 +1,369 @@
+        self.best_state = None
+
+        # Hass object
+        self.hass = hass
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    def update(self):
+        """Get the latest and forecasted weather from FMI."""
+
+        def update_best_weather_condition():
+            if self.hourly is None:
+                return
+
+            if self.current is None:
+                return
+
+            curr_date = date.today()
+
+            # Init values
+            self.best_state = BEST_CONDITION_NOT_AVAIL
+            self.best_time = self.current.data.time.astimezone(tz.tzlocal())
+            self.best_temperature = self.current.data.temperature.value
+            self.best_humidity = self.current.data.humidity.value
+            self.best_wind_speed = self.current.data.wind_speed.value
+            self.best_precipitation = self.current.data.precipitation_amount.value
+
+            for forecast in self.hourly.forecasts:
+                local_time = forecast.time.astimezone(tz.tzlocal())
+
+                if local_time.day == curr_date.day + 1:
+                    # Tracking best conditions for only this day
+                    break
+
+                if (
+                    (forecast.symbol.value in BEST_COND_SYMBOLS)
+                    and (forecast.wind_speed.value >= self.min_wind_speed)
+                    and (forecast.wind_speed.value <= self.max_wind_speed)
+                ):
+                    if (
+                        forecast.temperature.value >= self.min_temperature
+                        and forecast.temperature.value <= self.max_temperature
+                    ):
+                        if (
+                            forecast.humidity.value >= self.min_humidity
+                            and forecast.humidity.value <= self.max_humidity
+                        ):
+                            if (
+                                forecast.precipitation_amount.value >= self.min_precip
+                                and forecast.precipitation_amount.value
+                                <= self.max_precip
+                            ):
+                                # What more can you ask for?
+                                # Compare with temperature value already stored and update if necessary
+                                self.best_state = BEST_CONDITION_AVAIL
+
+                if self.best_state is BEST_CONDITION_AVAIL:
+                    if forecast.temperature.value > self.best_temperature:
+                        self.best_time = local_time
+                        self.best_temperature = forecast.temperature.value
+                        self.best_humidity = forecast.humidity.value
+                        self.best_wind_speed = forecast.wind_speed.value
+                        self.best_precipitation = forecast.precipitation_amount.value
+
+        # Current Weather
+        try:
+            self.current = fmi.weather_by_coordinates(self.latitude, self.longitude)
```

Comment:
You can use DataUpdateCoordinator for your API calls which removes the need for this try except logic and ensures less logging spam if the server is down.  See https://developers.home-assistant.io/docs/integration_fetching_data

Reply:
I tried this, but it was complicating the code. Configuration is common under a single domain key, but the entities have separate API calls. After trialing for some time, I decided to keep the original content, although I have combined the API calls under a single try/except block.

Reply:
I only see two different API calls. When using config entries, each entry will have only one set of coordinates (latitude and longitute).

Please use our coordinator helper and make one coordinator instance per API call. See the nsw integration for an example.